### ISSUE TITLE
False positive no-member in __traceback__.tb_frame

### ISSUE THREAD

**Issue by sam-s** [PyCQA/pylint#11218]
### Bug description

the code works fine, but pylint doesn't like it:

```python
try:
    foo()
except Exception as exn:
    last_tb = exn.__traceback__
    print(last_tb.tb_frame.f_locals)
```

### Command used

```shell
pylint --rcfile=/dev/null --disable=all --enable=no-member tbf.py
```

### Pylint output

```python
************* Module tbf
tbf.py:5:10: E1101: Class 'tb_frame' has no 'f_locals' member (no-member)
```

### Expected behavior

no diagnostics

### Pylint version

```shell
pylint 4.0.6
astroid 4.1.2
Python 3.14.4
```

### OS / Environment

ubuntu

**Comment by Pierre-Sassoulas** [comment]
Thank you for the report, I can reproduce it on the latest main (pylint 4.1.0-dev0 / astroid 4.2.0b3).

It could be an astroid bug: ``exn.__traceback__`` is correctly inferred as an instance of the (raw-built) builtins.traceback class, but its attributes are C-level data descriptors (member_descriptor), and astroid's raw_building.object_build_datadescriptor represents each of them as an empty nested ClassDef. So last_tb.tb_frame is inferred as an empty class  named tb_frame, and looking up f_locals on it fails. Look similar to https://github.com/pylint-dev/astroid/issues/580, which was worked around rather than fixing the raw building.
